aws/src/main/java/org/apache/iceberg/aws/RESTSigV4AuthManager.java

```diff
@@ -63,6 +63,7 @@ public RESTSigV4AuthSession catalogSession(
   @Override
   public RESTSigV4AuthSession contextualSession(
       SessionCatalog.SessionContext context, AuthSession parent) {
+    Preconditions.checkState(parent instanceof RESTSigV4AuthSession, "parent session is not SigV4");
```

Comment:
```suggestion
    Preconditions.checkState(parent instanceof RESTSigV4AuthSession, "Parent session is not SigV4: %s", parent);
```
same for the other check